pkg/jobs/pool.go

```diff
@@ -21,6 +21,8 @@ type Pool struct {
 	cronJobsMapLock         sync.Mutex
 	detachedCronJobsMapLock sync.Mutex
 	poolStatusLock          sync.Mutex
+	backoffLock             sync.Mutex
```

Comment:
Maybe. I polled Jason and Shane (I guess I forgot you) when I did this, and was told "1 mutex per variable". I was afraid if I used the same mutex for multiple variables I might get into a condition where read/write of a var might be locked unnecessarily long because another var was updating with the same lock.